src/server/system_services/account_server.js

```diff
@@ -799,7 +822,11 @@ async function update_external_connection(req) {
         },
         $set: {
             "sync_credentials_cache.$.access_key": identity,
-            "sync_credentials_cache.$.secret_key": encrypted_secret
+            "sync_credentials_cache.$.secret_key": encrypted_secret,
+            "sync_credentials_cache.$.azure_log_access_keys.azure_tenant_id": req.rpc_params.azure_log_access_keys.azure_tenant_id || undefined,
```

Comment:
Can you please check the behaviour when these fields are set to `undefined`? Actually I think I noticed a fluke with our `encode_json` which iterates over all the keys and captures the keys even if the values are set to `undefined`. I am not sure if that applies here.